openconnect-gp-okta

```diff
@@ -20,10 +20,16 @@ def extract_form(html):
 
 def prelogin(s, gateway):
     r = check(s.post('https://{}/ssl-vpn/prelogin.esp'.format(gateway)))
-    saml_req_html = base64.b64decode(lxml.etree.fromstring(r.content).find('saml-request').text)
-    saml_req_url, saml_req_data = extract_form(saml_req_html)
-    assert 'SAMLRequest' in saml_req_data
-    return saml_req_url + '?' + urllib.parse.urlencode(saml_req_data)
+    xml = lxml.etree.fromstring(r.content)
+    sam = xml.find('saml-auth-method').text
+    sr = base64.b64decode(xml.find('saml-request').text)
+    if sam == 'POST':
+        saml_req_url, saml_req_data = extract_form(sr)
+    elif sam == 'REDIRECT':
+        saml_req_url = sr, None
```

Comment:
Thanks for explaining this method; it makes more sense now. I think my comment still applies though -- you're not setting saml_req_data here at all. I think you meant to write

    saml_req_url, saml_req_data = sr, None

?

Reply:
Derrrrrp… sorry I didn't test and catch this sooner. Thanks. :man_facepalming: